packages/bot/src/stores/flyout-store.js

```diff
@@ -0,0 +1,202 @@
+/* eslint-disable no-underscore-dangle */
+import React                  from 'react';
+import { observable, action } from 'mobx';
+import FlyoutBlock            from '../components/flyout-block.jsx';
+
+export default class FlyoutStore {
+    block_listeners            = [];
+    block_workspaces           = [];
+    flyout_min_width           = 400;
+
+    @observable flyout_content = [];
+    @observable flyout_width   = this.flyout_min_width;
+    @observable is_visible     = false;
+
+    /**
+     * Parses XML contents passed by Blockly.Toolbox. Supports all default
+     * Blockly.Flyout elements i.e. <block>, <label>, <button> in their
+     * original format, e.g. <label text="Hello World" />
+     * @param {Element[]} xmlList list of XML nodes
+     * @memberof FlyoutStore
+     */
+    @action.bound setContents(xmlList) {
+        this.block_listeners.forEach(listener => Blockly.unbindEvent_(listener));
+        this.block_workspaces.forEach(workspace => workspace.dispose());
+        this.block_listeners = [];
+        this.block_workspaces = [];
+
+        const flyout_components = [];
+
+        xmlList.forEach((node, index) => {
```

Comment:
@aaron-binary i think this need to be handle inside your component. we dont do markup in stores.